From f32d02acf759c2d981390299f78936cab6fdc8b1 Mon Sep 17 00:00:00 2001
From: LuHui <luhux76@gmail.com>
Date: Thu, 16 Feb 2023 22:48:26 +0800
Subject: [PATCH 3/6] hw/block/m25p80: add flash id w25q128

Signed-off-by: LuHui <luhux76@gmail.com>
---
 hw/block/m25p80.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/block/m25p80.c b/hw/block/m25p80.c
index 802d2eb021..f7f65e0f9d 100644
--- a/hw/block/m25p80.c
+++ b/hw/block/m25p80.c
@@ -346,6 +346,7 @@ static const FlashPartInfo known_devices[] = {
     { INFO("w25q64",      0xef4017,      0,  64 << 10, 128, ER_4K) },
     { INFO("w25q80",      0xef5014,      0,  64 << 10,  16, ER_4K) },
     { INFO("w25q80bl",    0xef4014,      0,  64 << 10,  16, ER_4K) },
+    { INFO("w25q128",     0xef4018,      0,  64 << 10, 256, ER_4K) },
     { INFO("w25q256",     0xef4019,      0,  64 << 10, 512, ER_4K),
       .sfdp_read = m25p80_sfdp_w25q256 },
     { INFO("w25q512jv",   0xef4020,      0,  64 << 10, 1024, ER_4K),
-- 
2.39.1

