From 14ac69e7c831ece68887e9d9e3c026a445e0e36c Mon Sep 17 00:00:00 2001
From: LuHui <luhux76@gmail.com>
Date: Mon, 30 Jan 2023 22:22:05 +0800
Subject: [PATCH 7/7] hw/mips/vocore2.c: fill unimp memory region after
 machine.ram

Signed-off-by: LuHui <luhux76@gmail.com>
---
 hw/mips/vocore2.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/hw/mips/vocore2.c b/hw/mips/vocore2.c
index e9662224fd..1f3ea0658b 100644
--- a/hw/mips/vocore2.c
+++ b/hw/mips/vocore2.c
@@ -21,6 +21,7 @@
 #include "hw/mips/mips.h"
 #include "hw/mips/cpudevs.h"
 #include "hw/mips/mt7628.h"
+#include "hw/misc/unimp.h"
 #include "sysemu/sysemu.h"
 #include "hw/boards.h"
 #include "hw/mips/bios.h"
@@ -149,6 +150,17 @@ static void vocore2_init(MachineState *machine)
     memory_region_add_subregion(get_system_memory(),
                                 mt7628->memmap[MT7628_DEV_DDR],
                                 machine->ram);
+    /*
+     * fill a empty memory region,
+     * let uboot 'get_ram_size' function working.
+     * in uboot source tree : common/memsize.c
+     */
+    if (machine->ram_size < 256 * MiB) {
+        create_unimplemented_device("ddr(empty)",
+                                    mt7628->memmap[MT7628_DEV_DDR] +
+                                    machine->ram_size,
+                                    (256 * MiB) - machine->ram_size);
+    }
 
     /* Load kernel to RAM & goto kernel */
     reset_info = g_new0(ResetData, 1);
-- 
2.39.1

