From d6a7c16bd50303b26e9061416f5c4a199e41a912 Mon Sep 17 00:00:00 2001
From: LuHui <luhux76@gmail.com>
Date: Sun, 19 Feb 2023 11:13:06 +0800
Subject: [PATCH 8/9] hw/mips/mt7628: boot from flash

Signed-off-by: LuHui <luhux76@gmail.com>
---
 hw/mips/mt7628.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/hw/mips/mt7628.c b/hw/mips/mt7628.c
index f5df7413c0..391210b0eb 100644
--- a/hw/mips/mt7628.c
+++ b/hw/mips/mt7628.c
@@ -321,11 +321,21 @@ static void mt7628_board_init(MachineState *machine)
                                     machine->ram_size,
                                     (256 * MiB) - machine->ram_size);
     }
-    /* Load kernel to RAM & goto kernel */
+
+    /*
+     * PC register maybe at flash direct access address, after bootrom.
+     * I hexdump the mainline u-boot-spl.bin,
+     * begin of flash 4 bytes is a instruction use for jump.
+     * NOTE: uboot-with-spl.bin will loop at ddr_calibrate.
+     * TODO: emulate ddr_calibrate?.
+     * No, you can simple skip it, use your hex editor.
+     * qemu's DDR is always working, don't need care about it.
+     */
     reset_info = g_new0(ResetData, 1);
     reset_info->cpu = mt7628->cpu;
-    reset_info->vector = reset_info->cpu->env.active_tc.PC;
+    reset_info->vector = (target_long)(int32_t)0x9c000000;
     qemu_register_reset(main_cpu_reset, reset_info);
+    /* Load kernel to RAM & goto kernel */
     if (kernel_filename) {
         loaderparams.ram_size = machine->ram_size;
         loaderparams.kernel_filename = kernel_filename;
-- 
2.39.1

